trieste/models/config.py

```diff
@@ -21,93 +21,135 @@
 from __future__ import annotations
 
 from dataclasses import dataclass, field
-from typing import Any, Dict, Iterable, Tuple, Type, Union
+from typing import Any, Dict, Iterable, Type, Union
 from warnings import warn
 
+import gpflow
 import tensorflow as tf
 
 from .interfaces import TrainableProbabilisticModel
-from .optimizer import Optimizer
+from .optimizer import BatchOptimizer, Optimizer
 
 
 class ModelRegistry:
```

Comment:
Is there a reason to make this one registry rather than two (ModelRegistry, OptimizerRegistry)? Separation of concerns would suggest splitting them.

Reply:
yeah, I was thinking about it, but these optimizers are for models so finally opted for a single registry, but if you think its better to separate them, we can do that

Reply:
I'm not too fussed.